qemu/tests/check_nic_link_status.py

```diff
@@ -0,0 +1,100 @@
+import time
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_net
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    KVM guest link test:
+    1) Boot up guest with one nic
+    2) Disable guest link by set_link before guest os boot up
+    3) Check guest nic operstate status
+    4) Reboot the guest, then check guest nic operstate
+    5) Re-enable guest link by set_link
+    6) Check guest nic operstate and ping host from guest
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def guest_interface_operstate_check(session, expect_status):
+        """
+        Check Guest interface operstate
+        """
+        if params.get("os_type") == "linux":
+            guest_ifname = utils_net.get_linux_ifname(session, vm.get_mac_address())
+            if_operstate = utils_net.get_net_if_operstate(guest_ifname,
+                                                          session.cmd_output_safe)
+        else:
+            if_operstate = utils_net.get_windows_nic_attribute(session,
+                                                               "macaddress",
+                                                               vm.get_mac_address(),
+                                                               "netconnectionstatus")
+
+        if if_operstate != expect_status:
+            err_msg = "Guest interface %s status error, " % guest_ifname
+            err_msg += "currently interface status is '%s', " % if_operstate
+            err_msg += "but expect status is '%s'" % expect_status
+            test.fail(err_msg)
+        test.log.info("Guest interface operstate '%s' is exactly as expected",
+                      if_operstate)
+
+    def set_link_test(linkid, link_up):
+        """
+        Issue set_link commands and test its function
+
+        :param linkid: id of devices to be tested
+        :param link_up: flag linkid is up or down
+
+        """
+        vm.set_link(linkid, up=link_up)
+        time.sleep(1)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    device_id = vm.virtnet[0].device_id
+    device_mac = vm.virtnet[0].mac
+    params.get("paused_after_start_vm") == "yes"
+    set_link_test(device_id, False)
+    vm.resume()
+    login_timeout = float(params.get("login_timeout", 360))
+    session = vm.wait_for_serial_login(timeout=login_timeout)
+    # Win guest '2' represent 'Connected', '7' represent 'Media disconnected'
+    os_type = params.get("os_type")
+    if os_type == "linux":
+        expect_down_status = params.get("down-status", "down")
+        expect_up_status = params.get("up-status", "up")
+    else:
+        expect_down_status = params.get("win_media_disconnected", "7")
+        expect_up_status = params.get("win_media_connected", "2")
```

Comment:
I think using the same parameter name is okay.

Reply:
> I think using the same parameter name is okay.

Could you please help provide more info？ I did not got your point.